### ISSUE TITLE
Inference needs three processes, not one — the supervisor manages one

### ISSUE THREAD

**Issue by Rumeasiyan** [Rumeasiyan/askwell#89]
Found while verifying `M0-MODEL-BE-019` (#88).

## What

`M0-MODEL-DEPLOY-018` says the native process "serves generation, embeddings and reranking from the same process". **It cannot.**

Measured against `llama.cpp` b10645 with `unsloth/Qwen3.5-4B-GGUF Q4_K_M`:

| | |
| --- | --- |
| generation | works |
| embeddings | works — but only with `--embeddings --pooling mean`, and the vectors are **2560 dimensions** |
| reranking | `501 This server does not support reranking. Start it with --reranking` |

## Why it matters

**The dimension is the real problem.** `docs/architecture.md` §6 puts retrieval on `bge-m3` at **1024** dimensions, and `chunks.embedding` is `vector(1024)` by migration. A generation model's hidden state is neither that width nor trained for retrieval — so embeddings from the generation process are not merely suboptimal, they will not insert.

`--reranking` needs a reranker model and is mutually exclusive with generation on one process.

So Askwell needs **three** native processes: generation, embeddings (`bge-m3`), reranking. The supervisor manages one.

## Not urgent, but blocking M1

Retrieval is where this bites. Nothing in M0 uses embeddings or reranking — the client paths are implemented and unit-tested, and generation is verified end to end.

## What to do

Extend `deploy/inference/askwell-inference` to supervise a set of processes rather than one, each with its own model, port and readiness. The state file becomes a map. `M7`'s installer then provisions three model files rather than one.

Refs #86, #88, `docs/architecture.md` §6

**Comment by Rumeasiyan** [comment]
Done in #97. M1 retrieval is unblocked.

```
generation  ready   Qwen3.5-4B-Q4_K_M.gguf
embedding   ready   bge-m3-FP16.gguf               1024 dimensions
reranking   ready   bge-reranker-v2-m3-FP16.gguf
```

Through the client, from inside the API container, over **one socket**:

```
generate: 8 tokens
embed:    2 vectors of 1024 dimensions
rerank:   "how much notice to end the lease?"
  -2.419  Either party may terminate on ninety days written notice.
  -7.846  The tenant shall pay rent monthly in advance.
 -11.028  The premises shall be used for office purposes only.
```

The reranker puts the right passage first by a wide margin — the signal retrieval needs.

**The scores are raw logits, not probabilities.** Negative and unbounded, so only their order means anything. The client returns them sorted and deliberately does not invite a threshold on the number itself.

## Why this was necessary rather than tidy

Embeddings from the generation model were **2560 dimensions** where `chunks.embedding` is `vector(1024)`. Not merely poor for retrieval — **the database would have refused the insert.**

## Design

One socket, three processes. The bridge routes by path prefix, which is already in the first line it reads, so containers never learn how many processes there are or on which ports. Generation is the default for anything unrecognised rather than an error — llama.cpp serves far more than these three endpoints, and refusing `/health` or `/props` would break them for no benefit.

Each role is supervised **independently**. One model missing does not stop the others: a user with no reranker can still ask questions, and telling them the assistant is down because retrieval cannot rank would be false.

Three supervisors share one state file, written under a lock — without it they read-modify-write over each other and a role's entry vanishes, which the API would read as that role never having reported.

A state file in the **single-process shape** — which every M0 install wrote — still reads, as "no roles" rather than as a failure. Otherwise the first launch after an upgrade would report the assistant missing.

## Constraint C9

Both models verified against the registry **before their names were written down**:

| | |
| --- | --- |
| `BAAI/bge-m3` | MIT, ungated, 36M downloads |
| `BAAI/bge-reranker-v2-m3` | Apache-2.0, ungated, 18M downloads |

Both permit redistribution and commercial use, which the offline installer in Phase 7 needs.

A test asserts the supervisor's role list matches `ROLES`. The supervisor is stdlib-only by necessity and cannot import it, so nothing else would catch them drifting.

**Version:** `0.2.1`.
